dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -552,7 +556,20 @@ CFileDefExt::InitFileAttr(HWND hwndDlg)
             FileSize.u.LowPart = FileInfo.nFileSizeLow;
             FileSize.u.HighPart = FileInfo.nFileSizeHigh;
             if (SH_FormatFileSizeWithBytes(&FileSize, wszBuf, _countof(wszBuf)))
+            {    
                 SetDlgItemTextW(hwndDlg, 14011, wszBuf);
+                // Calculate size on disc
+                if( GetVolumePathName(m_wszPath, lpszVolumePathName, sizeof(lpszVolumePathName) / sizeof(TCHAR)) ) {
+                    if( GetDiskFreeSpace(lpszVolumePathName, &lpSectorsPerCluster, &lpBytesPerSector, NULL, NULL) ) {
```

Comment:
`GetDiskFreeSpaceEx` is better.

Reply:
No, the right solution is to use DeviceIoControl with IOCTL_DISK_GET_DRIVE_GEOMETRY. I add this to the TODO list

Reply:
The shell32 is not supposed to call like that these sort of DeviceIoControl things...

Reply:
What is your prefered solution ? GetDiskFreeSpace as it is or DeviceIoControl ? I see no other option to retrieve the disk geometry elements required to to the necessary computation.

Reply:
When you have a file in a filesystem volume that spans many partitions on different disks, how are you going to just use IOCTL_DISK_GET_DRIVE_GEOMETRY to get the correct information?

Reply:
I don't know. Maybe we should stay with the simple GetDiskFreeSpace implementation, and log a limitation on complex FS over multiple disk (which may have other impact on ROS code anyway)
Your advice ?

Reply:
See https://docs.microsoft.com/en-us/previous-versions/technet-magazine/hh148159(v=msdn.10) for why it's not trivial.

Reply:
OK, this is a definitive answer. Let's keep the simple implementation as it is and document in code the limitation.